src/isomorphic/children/sliceChildren.js

```diff
@@ -32,16 +32,16 @@ function sliceChildren(children, start, end) {
   var slicedChildren = {};
   var flattenedMap = flattenChildren(children);
   var ii = 0;
+  var endIsDefined = end != null;
   for (var key in flattenedMap) {
     if (!flattenedMap.hasOwnProperty(key)) {
       continue;
     }
-    var child = flattenedMap[key];
     if (ii >= start) {
-      slicedChildren[key] = child;
+      slicedChildren[key] = flattenedMap[key];
     }
     ii++;
-    if (end != null && ii >= end) {
```

Comment:
Yeah, I'm not sure how I feel about setting end to a magic number either.  Both solutions seem kinda sucky because they add complexity, I'd rather just keep this inline null check I think.  This is the type of thing that really should be handled by the JIT rather than hand-optimized.  Given that this is the type of thing that a JIT could easily optimize, do we even know that this improves perf?

I'm inclined to agree with @syranide on this one.  But @sebmarkbage has a lower threshold for acceptable perf optimizations than I do, so tagging him in case he wants to jump in.